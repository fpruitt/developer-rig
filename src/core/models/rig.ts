import { ExtensionManifest } from './manifest';
import { SegmentMap } from '../../util/api';

export interface RigProject {
  extensionViews: RigExtensionView[],
  projectFolderPath: string;
  manifest: ExtensionManifest;
  secret: string;
  frontendFolderName: string;
  frontendCommand: string;
  backendCommand: string;
}

export interface RigExtensionView {
  x: number;
  y: number;
  orientation: string;
  id: string;
  channelId: string;
  features: {
    isChatEnabled: boolean;
  };
  type: string;
  mode: string;
  role: string;
  linked: boolean;
  linkedUserId: string;
  opaqueId: string;
  isPopout: boolean;
  frameSize: FrameSize;
}

export interface FrameSize {
  height: number;
  width: number;
}

export interface ChannelSegments {
  [channelId: string]: SegmentMap;
}

export interface Configurations {
  globalSegment: ExtensionCoordinator.Segment;
  channelSegments: ChannelSegments;
}
