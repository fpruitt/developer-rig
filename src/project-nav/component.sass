.project-nav
  background-color: #392f5b
  display: grid
  grid-auto-columns: min-content
  grid-auto-flow: column

  &__link
    color: #c5b5e1
    font-size: 15px
    height: 40px
    line-height: 40px
    text-align: center
    text-decoration: none
    width: 140px

    &--active
      background-color: black
      color: white

  &__button
    background-color: #392f5b
    border: 0
    color: white
    font-size: 15px
    height: 40px
    cursor: pointer
    width: 140px
