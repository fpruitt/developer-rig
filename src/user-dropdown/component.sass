=transition($properties)
  -moz-transition: $properties
  -ms-transition: $properties
  -o-transition: $properties
  -webkit-transition: $properties
  transition: $properties

.user-dropdown
  display: flex
  flex-direction: column
  align-items: center
  font-size: 14px
  color: white
  text-align: center
  vertical-align: middle
  height: 100%

  &__tab
    align-items: center
    display: flex
    flex-direction: row
    height: 100%
    justify-content: space-evenly
    width: 100%

  &__image
    width: 30px
    height: 30px
    margin: 5px

  &__triangle
    color: #fff
    height: 10px
    width: 10px
    transform: rotate(180deg)
    +transition(transform 0.3s ease)

    &--open
      transform: rotate(0deg)

  &__menu
    position: absolute
    text-align: left
    margin-top: 40px
    font-size: 12px
    color: white
    background-color: rgba(32, 29, 44, 0.8)
    width: 10rem
    height: 0
    visibility: hidden
    +transition(height 0.3s ease)
    z-index: 1000
    >ul
      >li
        margin-top: 0.5rem
        >a
          color: white
          text-decoration: none
          >img
            margin-left: .5em

      list-style: none
      margin: 1rem
      padding: 0

  &__menu--open
    height: 9rem
    visibility: visible

  &__divider
    width: 100%
    border-top: 1px solid grey
