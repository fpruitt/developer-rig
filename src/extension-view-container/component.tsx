import * as React from 'react';
import './component.sass';
import classNames = require('classnames');
import { Configurations } from '../core/models/rig';
import { Console } from '../console';
import { ExtensionMode } from '../constants/extension-coordinator';
import { ExtensionView } from '../extension-view';
import { ExtensionViewButton } from '../extension-view-button';
import { RigExtensionView } from '../core/models/rig';
import { ExtensionManifest } from '../core/models/manifest';
import { createExtensionObject } from '../util/extension';
import { ExtensionViewDialog, ExtensionViewDialogState } from '../extension-view-dialog';
import { EditViewProps, EditViewDialog } from '../edit-view-dialog';

interface Props {
  configurations: Configurations;
  extensionViews: RigExtensionView[];
  isDisplayed: boolean;
  manifest: ExtensionManifest;
  secret: string;
  createExtensionViewHandler: (extensionViewDialogState: ExtensionViewDialogState) => Promise<void>;
  deleteExtensionViewHandler: (id: string) => void;
  editViewHandler: (viewForEdit: RigExtensionView, newViewState: EditViewProps) => void;
}

interface State {
  mockTriggersEnabled: boolean;
  showingExtensionsViewDialog: boolean;
  viewForEdit?: RigExtensionView;
}

const ConfigNames: { [key: string]: string; } = {
  [ExtensionMode.Config]: 'Broadcaster Configuration',
  [ExtensionMode.Dashboard]: 'Broadcaster Live Dashboard',
};

export class ExtensionViewContainer extends React.Component<Props, State> {
  public state: State = {
    mockTriggersEnabled: false,
    showingExtensionsViewDialog: false,
  };

  static getDerivedStateFromProps(props: Props, _state: State): Partial<State> {
    if (!props.isDisplayed) {
      return { showingExtensionsViewDialog: false, viewForEdit: null };
    }
    return null;
  }

  private toggleMockTriggers = () => {
    this.setState((previousState) => ({ mockTriggersEnabled: !previousState.mockTriggersEnabled }));
  }

  private constructConfiguration(channelId: string): ExtensionCoordinator.Configuration {
    const configuration: ExtensionCoordinator.Configuration = {};
    const { configurations: { globalSegment, channelSegments } } = this.props;
    if (globalSegment) {
      configuration.global = globalSegment;
    }
    const segments = channelSegments[channelId];
    if (segments) {
      if (segments.broadcaster) {
        configuration.broadcaster = segments.broadcaster;
      }
      if (segments.developer) {
        configuration.developer = segments.developer;
      }
    }
    return configuration;
  }

  private openEditViewDialog = (id: string) => {
    this.setState({
      viewForEdit: this.props.extensionViews.filter((extensionView) => extensionView.id === id)[0],
    });
  }

  private saveView = (viewForEdit: RigExtensionView, newViewState: EditViewProps) => {
    this.props.editViewHandler(viewForEdit, newViewState);
    this.closeEditViewDialog();
  }

  private closeEditViewDialog = () => {
    this.setState({ viewForEdit: null });
  }

  private openExtensionViewDialog = () => {
    this.setState({ showingExtensionsViewDialog: true });
  }

  private createExtensionView = async (extensionViewDialogState: ExtensionViewDialogState) => {
    await this.props.createExtensionViewHandler(extensionViewDialogState);
    this.closeExtensionViewDialog();
  }

  private closeExtensionViewDialog = () => {
    this.setState({ showingExtensionsViewDialog: false });
  }

  public render() {
    let extensionViews: JSX.Element[] = [];
    if (this.props.extensionViews && this.props.extensionViews.length > 0) {
      extensionViews = this.props.extensionViews.map((view, index) => {
        const linkedUserId = view.linked ? view.linkedUserId : '';
        const extension = createExtensionObject(this.props.manifest, index.toString(), view.role,
          linkedUserId, view.channelId, this.props.secret, view.opaqueId);
        const configuration = this.constructConfiguration(view.channelId);
        return (
          <ExtensionView
            key={view.id}
            view={view}
            configuration={configuration}
            extension={extension}
            role={view.mode === ExtensionMode.Viewer ? view.role : ConfigNames[view.mode]}
            openEditViewHandler={this.openEditViewDialog}
            deleteViewHandler={this.props.deleteExtensionViewHandler}
            mockApiEnabled={this.state.mockTriggersEnabled}
          />
        );
      });
    }
    const [switchClassName, handleClassName, checkClassName] = ['', '-handle', '-check'].map((suffix) => {
      const root = "trigger-bar__switch" + suffix;
      return classNames(root, {
        [`${root}--on`]: this.state.mockTriggersEnabled,
      });
    });
    const wrapperClassName = classNames('view-container-wrapper', {
      'view-container-wrapper--hidden': !this.props.isDisplayed,
    });
    return (
      <>
        <div className={wrapperClassName}>
          <div className="trigger-bar">
            <svg version="1.1" viewBox="0 0 44 20" height="20" width="44">
              <g onClick={this.toggleMockTriggers}>
                <rect className={switchClassName} rx="2" ry="2" x="0.5" y="0.5" height="19" width="43" />
                <path className={checkClassName} d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z" />
                <rect className={handleClassName} rx="2" ry="2" x="0.5" y="0.5" height="19" width="21" />
              </g>
            </svg>
            <div className="trigger-bar__text">Use Mock Triggers</div>
          </div>
          <div className="view-container">
            {extensionViews}
          </div>
          <div>
            <ExtensionViewButton onClick={this.openExtensionViewDialog} />
          </div>
          <Console />
        </div>
        {this.state.showingExtensionsViewDialog && (
          <ExtensionViewDialog
            extensionViews={this.props.manifest.views}
            closeHandler={this.closeExtensionViewDialog}
            saveHandler={this.createExtensionView}
          />
        )}
        {this.state.viewForEdit && (
          <EditViewDialog
            viewForEdit={this.state.viewForEdit}
            closeHandler={this.closeEditViewDialog}
            saveViewHandler={this.saveView}
          />
        )}
      </>
    );
  }
}
