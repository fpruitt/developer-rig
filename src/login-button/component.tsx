import * as React from 'react';
import './component.sass';

export class LoginButton extends React.Component<{}>{
  public render() {
    const rigAuthUrl = 'https://id.twitch.tv/oauth2/authorize?client_id=7db8cgf6p0b43n3yzwfu6i04hp8f4b&redirect_uri=http://localhost:3000&response_type=token&scope=user:edit+extensions:edit:catalog';
    return (
        <a href={rigAuthUrl}>
          <button className='login-button'>Sign In</button>
        </a>
    );
  }
}
