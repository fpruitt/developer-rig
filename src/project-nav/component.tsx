import * as React from 'react';
import { NavLink } from 'react-router-dom';
import './component.sass';
import { RigProject } from '../core/models/rig';
import { NavItem } from '../constants/nav-items';

export interface Props {
  projects: RigProject[];
  createProject: () => void;
}

export class ProjectNav extends React.Component<Props>{
  public render() {
    return (
      <div className="project-nav">
        {this.props.projects.map((project, index) => (
          <NavLink key={index} className="project-nav__link" activeClassName="project-nav__link--active" to={`/${index}${NavItem.ProjectOverview}`}>{project.manifest.name}</NavLink>
        ))}
        <button className="project-nav__button" onClick={this.props.createProject}>Add New Project</button>
      </div>
    );
  }
}
