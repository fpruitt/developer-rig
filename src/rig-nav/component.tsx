import * as React from 'react';
import * as classNames from 'classnames';
import './component.sass';
import { NavItem } from '../constants/nav-items';
import { UserDropdown } from '../user-dropdown';
import { UserSession } from '../core/models/user-session';
import { ExtensionManifest } from '../core/models/manifest';
import { NavLink } from 'react-router-dom';

export interface PublicProps {
  projectIndex: number,
  manifest: ExtensionManifest,
  deleteProject: () => void,
}

export interface ReduxStateProps {
  session?: UserSession,
  mockApiEnabled?: boolean,
}

type Props = PublicProps & ReduxStateProps;

export class RigNavComponent extends React.Component<Props> {
  public render() {
    const { session, manifest } = this.props;
    const productManagementClass = classNames('offset', 'top-nav-item', {
      'top-nav-item__disabled': !(session && session.login) || !(manifest && manifest.bitsEnabled),
    });
    const configurationServiceClass = classNames('offset', 'top-nav-item', {
      'top-nav-item__disabled': manifest && manifest.configurationLocation !== 'hosted',
    });
      return (
        <div className='top-nav'>
          <div className='personal-bar'>
            {manifest && <div className='personal-bar__ext-name'>
              <span>{manifest.name}</span>
              <button className="personal-bar__button" onClick={this.props.deleteProject}>Delete</button>
            </div>}
            {session && session.login && <div className='top-nav-item__login'>
              <UserDropdown session={session} />
            </div>}
          </div>
          <div className='top-nav__item-container'>
            <NavLink className="offset top-nav-item" to={`/${this.props.projectIndex}${NavItem.ProjectOverview}`} exact={true} activeClassName="top-nav-item__selected">
              Project Overview
            </NavLink>
            <NavLink className="offset top-nav-item" to={`/${this.props.projectIndex}${NavItem.ExtensionViews}`} exact={true} activeClassName="top-nav-item__selected">
              Extension Views
            </NavLink>
            <NavLink className={productManagementClass} to={`/${this.props.projectIndex}${NavItem.ProductManagement}`} exact={true} activeClassName="top-nav-item__selected">
              Manage Bits Products
            </NavLink>
            <NavLink className={configurationServiceClass} to={`/${this.props.projectIndex}${NavItem.ConfigurationService}`} exact={true} activeClassName="top-nav-item__selected">
              Configuration Service
            </NavLink>
          </div>
        </div>
      );
  }
}
