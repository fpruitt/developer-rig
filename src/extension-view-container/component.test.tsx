import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ChannelSegments } from '../core/models/rig';
import { createViewsForTest, createExtensionManifestForTest } from '../tests/constants/extension';
import { ExtensionViewContainer } from './component';
import { ExtensionAnchors } from '../constants/extension-types';
import { ViewerTypes } from '../constants/viewer-types';
import { ExtensionAnchor } from '../constants/extension-coordinator';

const segment = {
  content: 'content',
  version: 'version',
};

const setupShallow = setupShallowTest(ExtensionViewContainer, () => ({
  configurations: {
    globalSegment: segment,
    channelSegments: {
      twitch: {
        broadcaster: segment,
        developer: segment,
      },
    } as ChannelSegments,
  },
  extensionViews: createViewsForTest(0, '', ''),
  isDisplayed: true,
  manifest: createExtensionManifestForTest(),
  secret: '',
  createExtensionViewHandler: jest.fn().mockImplementation(() => Promise.resolve()),
  deleteExtensionViewHandler: jest.fn(),
  editViewHandler: jest.fn(),
}));

describe('<ExtensionViewContainer />', () => {
  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('opens, creates, and closes ExtensionViewDialog', async () => {
    const { wrapper } = setupShallow();
    wrapper.find('ExtensionViewButton').simulate('click');
    expect(wrapper.instance().state.showingExtensionsViewDialog).toBe(true);
    wrapper.update();
    expect(wrapper.find('ExtensionViewDialog')).toBeDefined();
    await wrapper.find('ExtensionViewDialog').get(0).props.saveHandler();
    return new Promise((resolve, reject) => {
      setTimeout(() => {
        try {
          expect(wrapper.instance().state.showingExtensionsViewDialog).toBe(false);
          expect(wrapper.instance().props.createExtensionViewHandler).toHaveBeenCalled();
          resolve();
        } catch (ex) {
          reject(ex.message);
        }
      });
    });
  });

  it('enables mock triggers', async () => {
    const { wrapper } = setupShallow();
    expect(wrapper.find('g')).toBeDefined();
    wrapper.find('g').get(0).props.onClick();
    expect(wrapper.instance().state.mockTriggersEnabled).toBe(true);
  });

  it('opens, saves, and closes EditViewDialog', () => {
    const extensionViews = createViewsForTest(1, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut);
    const { wrapper } = setupShallow({ extensionViews });
    wrapper.find('ExtensionView').get(0).props.openEditViewHandler('1');
    expect(wrapper.instance().state.viewForEdit).toBe(extensionViews[0]);
    wrapper.update();
    expect(wrapper.find('EditViewDialog')).toBeDefined();
    wrapper.find('EditViewDialog').get(0).props.saveViewHandler();
    expect(wrapper.instance().state.viewForEdit).toBeNull();
  });
});
