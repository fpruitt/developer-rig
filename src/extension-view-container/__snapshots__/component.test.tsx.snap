// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionViewContainer /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionViewContainer
    configurations={
        Object {
            "channelSegments": Object {
              "twitch": Object {
                "broadcaster": Object {
                  "content": "content",
                  "version": "version",
                },
                "developer": Object {
                  "content": "content",
                  "version": "version",
                },
              },
            },
            "globalSegment": Object {
              "content": "content",
              "version": "version",
            },
          }
    }
    createExtensionViewHandler={[Function]}
    deleteExtensionViewHandler={[Function]}
    editViewHandler={[Function]}
    extensionViews={Array []}
    isDisplayed={true}
    manifest={
        Object {
            "authorName": "test",
            "bitsEnabled": true,
            "configurationLocation": "custom",
            "description": "test",
            "hasChatSupport": false,
            "iconUrls": Object {
              "100x100": "test",
            },
            "id": "test",
            "name": "test",
            "requestIdentityLink": false,
            "sku": "test",
            "state": "test",
            "summary": "test",
            "vendorCode": "test",
            "version": "0.0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "canLinkExternalContent": false,
                "size": 1024,
                "viewerUrl": "https://test:8080",
                "zoom": false,
                "zoomPixels": 24,
              },
              "config": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "liveConfig": Object {
                "canLinkExternalContent": false,
                "viewerUrl": "https://test:8080",
              },
              "panel": Object {
                "canLinkExternalContent": false,
                "height": 300,
                "viewerUrl": "https://test:8080",
              },
            },
            "whitelistedConfigUrls": Array [],
            "whitelistedPanelUrls": Array [],
          }
    }
    secret=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="view-container-wrapper"
>
          <div
                    className="trigger-bar"
          >
                    <svg
                              height="20"
                              version="1.1"
                              viewBox="0 0 44 20"
                              width="44"
                    >
                              <g
                                        onClick={[Function]}
                              >
                                        <rect
                                                  className="trigger-bar__switch"
                                                  height="19"
                                                  rx="2"
                                                  ry="2"
                                                  width="43"
                                                  x="0.5"
                                                  y="0.5"
                                        />
                                        <path
                                                  className="trigger-bar__switch-check"
                                                  d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                                        />
                                        <rect
                                                  className="trigger-bar__switch-handle"
                                                  height="19"
                                                  rx="2"
                                                  ry="2"
                                                  width="21"
                                                  x="0.5"
                                                  y="0.5"
                                        />
                              </g>
                    </svg>
                    <div
                              className="trigger-bar__text"
                    >
                              Use Mock Triggers
                    </div>
          </div>
          <div
                    className="view-container"
          >
                    
          </div>
          <div>
                    <ExtensionViewButton
                              onClick={[Function]}
                    />
          </div>
          <Console />
</div>,
        false,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="trigger-bar"
>
              <svg
                            height="20"
                            version="1.1"
                            viewBox="0 0 44 20"
                            width="44"
              >
                            <g
                                          onClick={[Function]}
                            >
                                          <rect
                                                        className="trigger-bar__switch"
                                                        height="19"
                                                        rx="2"
                                                        ry="2"
                                                        width="43"
                                                        x="0.5"
                                                        y="0.5"
                                          />
                                          <path
                                                        className="trigger-bar__switch-check"
                                                        d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                                          />
                                          <rect
                                                        className="trigger-bar__switch-handle"
                                                        height="19"
                                                        rx="2"
                                                        ry="2"
                                                        width="21"
                                                        x="0.5"
                                                        y="0.5"
                                          />
                            </g>
              </svg>
              <div
                            className="trigger-bar__text"
              >
                            Use Mock Triggers
              </div>
</div>,
            <div
              className="view-container"
>
              
</div>,
            <div>
              <ExtensionViewButton
                            onClick={[Function]}
              />
</div>,
            <Console />,
          ],
          "className": "view-container-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <svg
                  height="20"
                  version="1.1"
                  viewBox="0 0 44 20"
                  width="44"
>
                  <g
                                    onClick={[Function]}
                  >
                                    <rect
                                                      className="trigger-bar__switch"
                                                      height="19"
                                                      rx="2"
                                                      ry="2"
                                                      width="43"
                                                      x="0.5"
                                                      y="0.5"
                                    />
                                    <path
                                                      className="trigger-bar__switch-check"
                                                      d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                                    />
                                    <rect
                                                      className="trigger-bar__switch-handle"
                                                      height="19"
                                                      rx="2"
                                                      ry="2"
                                                      width="21"
                                                      x="0.5"
                                                      y="0.5"
                                    />
                  </g>
</svg>,
                <div
                  className="trigger-bar__text"
>
                  Use Mock Triggers
</div>,
              ],
              "className": "trigger-bar",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <g
                    onClick={[Function]}
>
                    <rect
                                        className="trigger-bar__switch"
                                        height="19"
                                        rx="2"
                                        ry="2"
                                        width="43"
                                        x="0.5"
                                        y="0.5"
                    />
                    <path
                                        className="trigger-bar__switch-check"
                                        d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                    />
                    <rect
                                        className="trigger-bar__switch-handle"
                                        height="19"
                                        rx="2"
                                        ry="2"
                                        width="21"
                                        x="0.5"
                                        y="0.5"
                    />
</g>,
                  "height": "20",
                  "version": "1.1",
                  "viewBox": "0 0 44 20",
                  "width": "44",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <rect
                        className="trigger-bar__switch"
                        height="19"
                        rx="2"
                        ry="2"
                        width="43"
                        x="0.5"
                        y="0.5"
/>,
                      <path
                        className="trigger-bar__switch-check"
                        d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
/>,
                      <rect
                        className="trigger-bar__switch-handle"
                        height="19"
                        rx="2"
                        ry="2"
                        width="21"
                        x="0.5"
                        y="0.5"
/>,
                    ],
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "trigger-bar__switch",
                        "height": "19",
                        "rx": "2",
                        "ry": "2",
                        "width": "43",
                        "x": "0.5",
                        "y": "0.5",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "rect",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "trigger-bar__switch-check",
                        "d": "m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "path",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "trigger-bar__switch-handle",
                        "height": "19",
                        "rx": "2",
                        "ry": "2",
                        "width": "21",
                        "x": "0.5",
                        "y": "0.5",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "rect",
                    },
                  ],
                  "type": "g",
                },
                "type": "svg",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Use Mock Triggers",
                  "className": "trigger-bar__text",
                },
                "ref": null,
                "rendered": "Use Mock Triggers",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "view-container",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ExtensionViewButton
                onClick={[Function]}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      false,
      undefined,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="view-container-wrapper"
>
            <div
                        className="trigger-bar"
            >
                        <svg
                                    height="20"
                                    version="1.1"
                                    viewBox="0 0 44 20"
                                    width="44"
                        >
                                    <g
                                                onClick={[Function]}
                                    >
                                                <rect
                                                            className="trigger-bar__switch"
                                                            height="19"
                                                            rx="2"
                                                            ry="2"
                                                            width="43"
                                                            x="0.5"
                                                            y="0.5"
                                                />
                                                <path
                                                            className="trigger-bar__switch-check"
                                                            d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                                                />
                                                <rect
                                                            className="trigger-bar__switch-handle"
                                                            height="19"
                                                            rx="2"
                                                            ry="2"
                                                            width="21"
                                                            x="0.5"
                                                            y="0.5"
                                                />
                                    </g>
                        </svg>
                        <div
                                    className="trigger-bar__text"
                        >
                                    Use Mock Triggers
                        </div>
            </div>
            <div
                        className="view-container"
            >
                        
            </div>
            <div>
                        <ExtensionViewButton
                                    onClick={[Function]}
                        />
            </div>
            <Console />
</div>,
          false,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="trigger-bar"
>
                <svg
                                height="20"
                                version="1.1"
                                viewBox="0 0 44 20"
                                width="44"
                >
                                <g
                                                onClick={[Function]}
                                >
                                                <rect
                                                                className="trigger-bar__switch"
                                                                height="19"
                                                                rx="2"
                                                                ry="2"
                                                                width="43"
                                                                x="0.5"
                                                                y="0.5"
                                                />
                                                <path
                                                                className="trigger-bar__switch-check"
                                                                d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                                                />
                                                <rect
                                                                className="trigger-bar__switch-handle"
                                                                height="19"
                                                                rx="2"
                                                                ry="2"
                                                                width="21"
                                                                x="0.5"
                                                                y="0.5"
                                                />
                                </g>
                </svg>
                <div
                                className="trigger-bar__text"
                >
                                Use Mock Triggers
                </div>
</div>,
              <div
                className="view-container"
>
                
</div>,
              <div>
                <ExtensionViewButton
                                onClick={[Function]}
                />
</div>,
              <Console />,
            ],
            "className": "view-container-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <svg
                    height="20"
                    version="1.1"
                    viewBox="0 0 44 20"
                    width="44"
>
                    <g
                                        onClick={[Function]}
                    >
                                        <rect
                                                            className="trigger-bar__switch"
                                                            height="19"
                                                            rx="2"
                                                            ry="2"
                                                            width="43"
                                                            x="0.5"
                                                            y="0.5"
                                        />
                                        <path
                                                            className="trigger-bar__switch-check"
                                                            d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                                        />
                                        <rect
                                                            className="trigger-bar__switch-handle"
                                                            height="19"
                                                            rx="2"
                                                            ry="2"
                                                            width="21"
                                                            x="0.5"
                                                            y="0.5"
                                        />
                    </g>
</svg>,
                  <div
                    className="trigger-bar__text"
>
                    Use Mock Triggers
</div>,
                ],
                "className": "trigger-bar",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <g
                      onClick={[Function]}
>
                      <rect
                                            className="trigger-bar__switch"
                                            height="19"
                                            rx="2"
                                            ry="2"
                                            width="43"
                                            x="0.5"
                                            y="0.5"
                      />
                      <path
                                            className="trigger-bar__switch-check"
                                            d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
                      />
                      <rect
                                            className="trigger-bar__switch-handle"
                                            height="19"
                                            rx="2"
                                            ry="2"
                                            width="21"
                                            x="0.5"
                                            y="0.5"
                      />
</g>,
                    "height": "20",
                    "version": "1.1",
                    "viewBox": "0 0 44 20",
                    "width": "44",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <rect
                          className="trigger-bar__switch"
                          height="19"
                          rx="2"
                          ry="2"
                          width="43"
                          x="0.5"
                          y="0.5"
/>,
                        <path
                          className="trigger-bar__switch-check"
                          d="m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z"
/>,
                        <rect
                          className="trigger-bar__switch-handle"
                          height="19"
                          rx="2"
                          ry="2"
                          width="21"
                          x="0.5"
                          y="0.5"
/>,
                      ],
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "trigger-bar__switch",
                          "height": "19",
                          "rx": "2",
                          "ry": "2",
                          "width": "43",
                          "x": "0.5",
                          "y": "0.5",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "rect",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "trigger-bar__switch-check",
                          "d": "m 6.5,9 4,4 6,-6 -1,-1 -5,5 -3,-3 z",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "path",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "trigger-bar__switch-handle",
                          "height": "19",
                          "rx": "2",
                          "ry": "2",
                          "width": "21",
                          "x": "0.5",
                          "y": "0.5",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "rect",
                      },
                    ],
                    "type": "g",
                  },
                  "type": "svg",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Use Mock Triggers",
                    "className": "trigger-bar__text",
                  },
                  "ref": null,
                  "rendered": "Use Mock Triggers",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "view-container",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ExtensionViewButton
                  onClick={[Function]}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        false,
        undefined,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
